packages/babel-plugin-transform-es2015-arrow-functions/test/fixtures/arrow-functions/spec/expected.js

```diff
@@ -14,7 +14,7 @@ function foo() {
 
     return function () {
       babelHelpers.newArrowCheck(this, _this2);
-      return this;
```

Comment:
Alright, I added support for spec arrow in subclass constructors, and made `this` stay `this` in spec arrows except for the subclass constructor case.